pkg/graveler/ref/commit_generation_priority_queue.go

```diff
@@ -0,0 +1,41 @@
+package ref
+
+import (
+	"github.com/treeverse/lakefs/pkg/graveler"
+)
+
+// CommitsGenerationPriorityQueue implements heap.Interface such that the commit with the greatest Generation value is
+// at the root of the heap.
+type CommitsGenerationPriorityQueue []*graveler.CommitRecord
+
+func NewCommitsGenerationPriorityQueue() CommitsGenerationPriorityQueue {
+	return make(CommitsGenerationPriorityQueue, 0)
+}
+
+func (c CommitsGenerationPriorityQueue) Len() int {
+	return len(c)
+}
+
+func (c CommitsGenerationPriorityQueue) Swap(i, j int) {
+	c[i], c[j] = c[j], c[i]
+}
+
+func (c *CommitsGenerationPriorityQueue) Push(x interface{}) {
+	rec := x.(*graveler.CommitRecord)
```

Comment:
If you're implementing your own structure, why accept an `interface{}` instead of `*CommitRecord`?

Reply:
Have to implement `heap.Interface` :(